pkg/splunk/enterprise/configuration.go

```diff
@@ -473,8 +473,8 @@ func updateSplunkPodTemplateWithConfig(client splcommon.ControllerClient, podTem
 		}
 	}
 
-	// Add custom volumes to splunk containers
-	if spec.Volumes != nil {
+	// Add custom volumes to splunk containers other than MC(where CR spec volumes are not needed)
+	if spec.Volumes != nil && instanceType != SplunkMonitoringConsole {
 		podTemplateSpec.Spec.Volumes = append(podTemplateSpec.Spec.Volumes, spec.Volumes...)
```

Comment:
If not now, later, can we have a test case for the MC, that validates the  MC Pod spec, to cover any accidental additions going forward? such that, we can take care of MC Pod for its uniqueness, as we keep adding more and more resources to the Pod spec.

Reply:
Yes, I agree there should be a test to verify the uniqueness of the MC pod. Will add that in going in future, need to verify all components we can add in the pod spec that can create an issue here

Reply:
As discussed, as a backlog item we can rephrase this API to **not** depend on **cr.spec** to make decisions on **PodTemplateSpec for MC**.